### PR TITLE
Code Review - Improvements and/or Changes

### PR DESCRIPTION
**Pull request by Rebelzob**
# Appoint Wheels
This project of @Rysth was made in collaboration with @Rebelzob , @IsmaelMastronardi and @SnakyMz 
The Project has two repositories:
- [**Frontend Link**](https://github.com/Rysth/REACT-AppointWheels)
- [**Backend Link**](https://github.com/Rysth/RAILS-AppointWheels)

# Things we have done:
## Backend
- [ ] Setup the database as per the ERD diagram
- [ ] Implement API only routes to handle JSON requests
- [ ] Added Documentation of the API using Rswag

## Frontend
- [ ] Added Login page and SignUp pages for user authentication
- [ ] Added protected routes to block non-user visits
- [ ] Added HomePage to display all available Cars for Rental
- [ ] Added Navigation bar for good UX
- [ ] Added CarDetails page to display information and option to rent selected Car
- [ ] Added NewCar page to add new Cars for Rental
- [ ] Added DeleteCar page to delete Cars from Rental
- [ ] Added Rentals page to display and delete rentals made by the specific user
- [ ] Added NewRental page for the user to create a new rental

## Additional features - Required only for teams with 5 members (optional for others)
- [ ] Installed Devise, JWT and Cancancan to implement authentication and authorization in the backend
- [ ] Implemented authorization for the NewCar and DeleteCar links to be only accessible by users who are admins

### CODE REVIEW

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Nice code organization ✔️ 
-  Good readme ✔️ 
-  Nice, beautiful frontend ✔️ 



#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

File: README.md [hunk 1]
```diff
@@ -95,8 +95,18 @@ Clone the repository:
 
 ### Usage <a name="usage"></a>
 
-- **npm start:** Run the app in development mode.
-- **npm run build:** Build the app for production.
+- Install the dependencies and start the server:
```
[the remark this answers is not in the document]
Comment (Rebelzob) [reply README.md:98]: @Meltrust We appreciate your feedback, it feels like you dug deep into the project and that makes us feel really good.

File: src/components/pages/car/carDetailForm.jsx [hunk 2]
```diff
@@ -8,7 +8,7 @@ import ReactDatePicker from 'react-datepicker';
 import { createRental } from '../../../redux/slices/rentalsSlice';
 import cities from '../../cities/city';
 
-function CarDetailForm({ car }) {
+const CarDetailForm = ({ car }) => {
```
Comment (Meltrust) [inline src/components/pages/car/carDetailForm.jsx:11]: - [ ] Good job adding validation for "new rentals" section.

However, in the car details section, I can still rent in the past:

image

image

## Suggested solution

You are using React Datepicker by Hackerone.

Therefore, this is your solution from their documentation:

https://reactdatepicker.com/#example-min-date

image

The `min-date` property will let you set a minimum date.  That way you can avoid reservations in the past. 

You can add the property here:

image

There is also an article that explains how to implement this:

https://medium.com/@ukaohachizoba6/how-to-disable-past-and-future-dates-with-react-date-picker-916f2fc6a9b7

image


Now that you know the solution and the way to implement it, kindly, make it so it is simply not possible to rent in the past, even from the car details page. 💪

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi,

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation, but you are almost there!






### To highlight:

-  Functions are now ES6✔️
-  Readme has been updated ✔️ 
-  Superfluous backend test folder has been removed✔️ 
-  Renting in the past has been fixed (only in "new rental" section) ✔️ 
-  It is possible now to access the admin functionality ✔️ 



#### You are really close to finishing the Microverse program!! Keep it up! 👍👍👍

You can do it

### After implementing the requested changes, please submit another review request. ♻️
_Check the comments under the review._







Cheers and Happy coding!👏👏👏

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the previous reviews unless it is requested otherwise._**

------

**Review by Meltrust — APPROVED** [review]
# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
#### _Unless you want to add more features, go ahead to your final presentation ⏩ ⏩ ⏩_ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- Beautiful, superb job ✔️  



Cheers and Happy coding!👏👏👏


------

**Pull request merged**